rom/dev/tests/test_fast_rom.rs

```diff
@@ -21,3 +22,29 @@ fn test_skip_kats() {
         false,
     );
 }
+
+#[test]
+fn test_fast_rom_production_error() {
+    let security_state =
+        *SecurityState::default().set_device_lifecycle(DeviceLifecycle::Production);
+
+    let rom = caliptra_builder::build_firmware_rom(&ROM_FAST_WITH_UART).unwrap();
+    let mut hw = caliptra_hw_model::new(BootParams {
+        init_params: InitParams {
+            rom: &rom,
+            security_state,
+            ..Default::default()
+        },
+        ..Default::default()
+    })
+    .unwrap();
+
+    // Let it run until a fatal error (should fail very early)
+    hw.step_until(|m| m.soc_ifc().cptra_fw_error_fatal().read() != 0);
```

Comment:
This will hang if no fatal error is reported instead of the test failing quickly. Is there a better way to do this or is that an acceptable way to fail?

Reply:
Consider checking for the actual error code.

Reply:
I do check for the specific error code (ROM_GLOBAL_FAST_ROM_IN_PRODUCTION) below with the assert. I just needed a condition to "step until" to know that was valid to check